services/httpd/handler.go

```diff
@@ -1087,13 +1089,23 @@ func gzipFilter(inner http.Handler) http.Handler {
 			inner.ServeHTTP(w, r)
 			return
 		}
-		gz := gzip.NewWriter(w)
-		defer gz.Close()
+		gz := gzipWriterPool.Get().(*gzip.Writer)
+		gz.Reset(w)
+		defer func() {
+			gz.Close()
+			gzipWriterPool.Put(gz)
```

Comment:
The way you have the code written right now is fine. I'd prefer to be a little more defensive, though. I think I'd like to see some utility functions instead. The ad-hoc inline usage is what concerns me. `getGzipWriter(io.Writer)` and `putGzipWriter(*gzip.Writer)`. That way, as long as those functions are used, someone would have to intentionally use the pool directly to re-use the old `ResponseWriter`